tests/zexperimental/test_mcmc.py

```diff
@@ -788,3 +788,38 @@ def test_find_valid_samplers_implicit_prior():
     valid_samplers = cuqi.experimental.mcmc.find_valid_samplers(target)
 
     assert(len(set(valid_samplers)) == 0)
+
+# ============ Testing of HybridGibbs ============
+
+def test_HybridGibbs_initial_point_setting():
+    """ Test that the HybridGibbs sampler adheres to the initial point set by sampling strategy. """
+
+    # Forward model
+    A, y_data, _ = cuqi.testproblem.Deconvolution1D(dim=10).get_components()
+
+    # Bayesian Problem
+    d = cuqi.distribution.Uniform(0, 100)
+    s = cuqi.distribution.Uniform(0, 100)
+    x = cuqi.distribution.Gaussian(0, lambda d: 1/d, geometry=A.domain_geometry)
+    y = cuqi.distribution.Gaussian(A@x, lambda s: 1/s)
+
+    # Joint distribution
+    joint = cuqi.distribution.JointDistribution(x, y, d, s)
+
+    # Posterior
+    posterior = joint(y=y_data)
+
+    # Sampling strategy
+    sampling_strategy = {
+        "d" : cuqi.experimental.mcmc.MH(initial_point=3),
```

Comment:
I just want to add a tiny comment that is not very relevant to this PR: initial point will be set to a `np.array` type if not specified, through [_get_default_initial_point](https://github.com/CUQI-DTU/CUQIpy/blob/4a07108f2bb3505dfa8b5db2a8b7ba1a1fde766d/cuqi/experimental/mcmc/_sampler.py#L389C1-L391C28); if specified, it will be set to whatever the user gives, a scalar in this case. I wonder if we should add something to check if the user specified initial point is compatible and possibly convert it to a uniform format, like `np.array`. 

A somehow related scenario, with the current test case, if we set the initial point of `x` to be a list, `list(0.5*np.ones(10))`, it will lead to an error:
```bash
--> 206 self.current_samples[par_name] = sampler.current_point.reshape(-1)

AttributeError: 'list' object has no attribute 'reshape'
```

Reply:
Yes this is a good point. I suggest we add an issue to handle setting initial point more robustly.

Reply:
#487